src/wp-includes/http.php

```diff
@@ -581,7 +581,21 @@ function wp_http_validate_url( $url ) {
 	}
 
 	$port = $parsed_url['port'];
-	if ( 80 === $port || 443 === $port || 8080 === $port ) {
+
+
+	/**
+	 * Controls list of ports considered safe in HTTP Api
+	 *
+	 * Allows to change and allow external requests for the HTTP request.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param array  $allowed_porst Array of integers for valid ports
+	 * @param string $host     Host name of the requested URL.
+	 * @param string $url      Requested URL.
+	 */
+	$allowed_ports = apply_filters( 'http_allowed_safe_ports', array( 80, 443, 8080 ), $host, $url );
```

Comment:
`$host` will be undefined when `$same_host` is `true`.

Suggest adding this at line 577:
```
} else {
    $host = $parsed_url['host'];
}
```